src/components/Tooltip/PTooltip.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <PPopOver ref="popover" class="p-tooltip" v-bind="{ placement, to }">
+    <template #target>
+      <span class="p-tooltip__target" @pointerenter="open">
+        <slot />
+      </span>
+    </template>
+    <slot name="tooltip">
+      <div class="p-tooltip__tooltip">
+        <div class="p-tooltip__content">
+          <slot name="content">
+            <p>{{ text }}</p>
+          </slot>
+        </div>
+      </div>
+    </slot>
+  </PPopOver>
+</template>
+
+<script lang="ts" setup>
+  import { computed, onUnmounted, ref } from 'vue'
+  import PPopOver from '@/components/PopOver/PPopOver.vue'
+  import { PositionMethod } from '@/types/position'
+  import { isDefined, isHtmlElement } from '@/utilities'
+  import { top, bottom, left, right } from '@/utilities/position'
+
+  const props = defineProps<{
+    text?: string,
+    placement?: PositionMethod | PositionMethod[],
+    to?: string | Element,
+  }>()
+
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-arguments
+  const popover = ref<InstanceType<typeof PPopOver>>()
```

Comment:
I have no idea why but eslint kept removing this type which would make it `Ref<any>` which isn't helpful.

Reply:
I used an a similar type and needed to instantiate it with `null`. So like `ref<InstanceType<typeof PPopOver> | null>(null)`, maybe the popover doesn't exist until some mounting event so it's not possible the const assignment is valid on instantiation?

Reply:
That did work. Thanks! Still doesn't make sense though. The ref doesn't exist until mounting but the type of a ref without a value is `ref<T | undefined>`. But least `null` gets rid of the linting issue.